src/awkward/contents/emptyarray.py

```diff
@@ -85,29 +82,26 @@ def copy(
         parameters=UNSET,
         backend=UNSET,
     ):
-        if not (parameters is UNSET or parameters is None or len(parameters) == 0):
-            deprecate(
-                f"{type(self).__name__} cannot contain parameters", version="2.2.0"
-            )
+        if not (parameters is UNSET or parameters is None):
+            raise ValueError(f"{type(self).__name__} cannot contain parameters")
         return EmptyArray(
-            parameters=self._parameters if parameters is UNSET else parameters,
             backend=self._backend if backend is UNSET else backend,
         )
 
     def __copy__(self):
         return self.copy()
 
     def __deepcopy__(self, memo):
-        return self.copy(parameters=copy.deepcopy(self._parameters, memo))
+        return self.copy()
 
     @classmethod
     def simplified(cls, *, parameters=None, backend=None):
-        if not (parameters is None or len(parameters) == 0):
```

Comment:
This is a more permissive test for the deprecations, but I think it's okay to break empty dicts here too.